Well i guess there can be a toggle in the respawn to spawn

crazyaccess opened the pull request.

JSH32 commented on src/main/resources/config.yml:
```diff
@@ -7,6 +7,9 @@ link:
   # Requires link to play on the server
   required: false
 
+  # Make the player respawn to spawn upon getting linked (disabled by default if the link required is set to false)
```
Make this say

```
# Should the player respawn at the world spawn after verification. This setting will only be used if link.required is true
```

JSH32 commented on src/main/resources/config.yml:
```diff
@@ -7,6 +7,9 @@ link:
   required: false
 
+  # Make the player respawn to spawn upon getting linked (disabled by default if the link required is set to false)
+  respawn_to_spawn: false
```
Change this to verify_spawn

JSH32 commented on src/main/java/com/github/riku32/discordlink/Config.java:
```diff
@@ -46,6 +46,7 @@
     private final String statusDeathUnlinked;
 
     private final boolean linkRequired;
+    private final boolean linkedRespawnToSpawn;
```
Change to verifySpawn

JSH32 commented on src/main/java/com/github/riku32/discordlink/discord/listeners/VerificationListener.java:
```diff
@@ -108,10 +108,13 @@ public void onButtonClick(ButtonClickEvent e) {
                                                     null, e.getUser().getAvatarUrl())
                                             .build())
                                             .queue();
+                                            return null;
```
I dont think this is needed so remove it

JSH32 commented on src/main/java/com/github/riku32/discordlink/discord/listeners/VerificationListener.java:
```diff
@@ -108,10 +108,13 @@ public void onButtonClick(ButtonClickEvent e) {
                             }
 
                             Bukkit.getScheduler().runTask(plugin, () -> {
+                                if (plugin.getPluginConfig().isLinkedRespawnToSpawn()) {
```
Make this isVerifySpawn

JSH32 commented on src/main/java/com/github/riku32/discordlink/Config.java:
```diff
@@ -55,6 +56,7 @@
      */
     public Config(FileConfiguration configuration) throws NoSuchElementException {
         linkRequired = Boolean.parseBoolean(getAsStringNotNull(configuration, "link.required"));
+        if (!linkRequired) { linkedRespawnToSpawn = Boolean.parseBoolean(getAsStringNotNull(configuration, "link.respawn_to_spawn")); };
```
Make this link.verify_spawn

JSH32 reviewed (commented):
Fix a few of the problems I lined out and then I'll approve it. 

Note: I would just make them myself but im in another state with a shitty laptop so im slow to these things till the 12th lmao. Doing code review on mobile rn lmao

Riku32 commented:
Also the action java version is wrong so im not gonna run it run. Dont worry abt that

Riku32 commented:
Also more descriptive names for PR and issues in the future please.

Riku32 commented:
1.11 btw

crazyaccess commented:
Noice

Riku32 commented:
Yeah fix the naming and the error i put in review and ill merge.

Riku32 commented:
Added two new reviews.

crazyaccess commented:
2 new reviews ?

Riku32 commented:
Comments. Read the things I just left

crazyaccess commented:
Oke i will have more descriptive stuff in the future

Riku32 commented:
Nono i mean the code review comments

Riku32 commented:
Where it says riku started review

crazyaccess commented:
I will test the artifact produced rn...

crazyaccess commented:
checked it myself with all the permutations and combinations (well there were just 4 permutations)

crazyaccess commented:
If you by chance merge this one, pls keep it open i will add more enhancement to this one like if the bed / spawn point (those nether spawn pillars) exist then they can optionally have a flag in the config on teleporting there instead of the world spawn or have a completely different teleport local altogether

crazyaccess committed:
Optional re spawn to world spawn option adding


Added initial else defaults to false for verifySpawn

that fucking coma

Maven with JDK 1.8 -> JDK 11 by Adopt

Update maven.yml

Emm.... Indentation was problem
Yeah reflected on the review and fixed the linked other way around... and that coma

Riku32 commented:
If you do that make another PR and make it required. Respawn point is ideal
